src/main/java/in/projecteka/consentmanager/consent/ConsentRequestRepository.java

```diff
@@ -47,7 +50,7 @@
     static {
         String s = "SELECT request_id, status, details, date_created, date_modified FROM consent_request " +
                 "where ";
-        SELECT_CONSENT_DETAILS_FOR_PATIENT = s + "patient_id=$1 and (status=$4 OR $4 IS NULL) " +
+        SELECT_CONSENT_DETAILS_FOR_PATIENT = s + "patient_id=$1 and status!=$5 and (status=$4 OR $4 IS NULL) " +
```

Comment:
why is this required? status != GRANTED? If you are filtering by status=$4, would it not get covered? 

also, is there a case ever that status will be null?

Reply:
Q: why is this required? status != GRANTED? If you are filtering by status=$4, would it not get covered?
A: This is required because we have to filter the requests by GRANTED and we have an ALL request param which return all status in order to filter it from ALL a separate clause is required. 

Q: is there a case ever that status will be null?
A: Yes, when we get 'ALL' as request param we make the status null from service after a check, basically null is ALL statuses.

Reply:
Q: how do you get only GRANTED requests then? 
Q? In the DB, will there be ever a case that the status is null? If so, why - even when the request is created - then it should say that the request is PENDING or something right?

Reply:
Q: how do you get only GRANTED requests then?
A: I believe we have a separate call for that - getConsentsByStatus

Q: In the DB, will there be ever a case that the status is null? If so, why - even when the request is created - then it should say that the request is PENDING or something right?
A: From what I could figure, when we insert a request into DB we insert as REQUESTED and then the status keeps changing when we GRANT or REVOKE or it Expires. So no, there isn't a case where in DB the status will be null.

Reply:
right .. so we don't need a null check. and please check if the getConsentsByStatus is redundant?

Reply:
Q: is there a case ever that status will be null?
A: Yes, when we get 'ALL' as request param we make the status null from service after a check, basically null is ALL statuses.
Like so, https://github.com/ProjectEKA/consent-manager/blob/CM-APP-43/src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java#L176

Also getConsentsByStatus is the only way to get GRANTED so no it's not redundant.

Reply:
ok. my question was whether the status in DB ever will be null. If you get STATUS as empty or null as part of request, you can just not include the status filter right? Unless you dont want to write another query at all.

Reply:
and we dont need the null check in the query - right?